src/main/scala/subsystem/HasTiles.scala

```diff
@@ -86,6 +91,49 @@ trait HasTileInterruptSources
   }
 }
 
+/** These are sources of "constants" that are driven into the tile.
+  * While they are not expected to change dyanmically while the tile is executing code,
+  * they may be either tied to a contant value or programmed during boot or reset.
+  * They need to be instantiated before tiles are attached to the subsystem containing them.
+  */
+trait HasTileInputConstants extends InstantiatesTiles { this: BaseSubsystem =>
+  // Collect dynamic hart id prefixes and/or distribute static hart ids to the tiles
+  val tileHartIdNode = BundleBridgeEphemeralNode[UInt]()
+  val tileHartIdNexusNode = BundleBridgeNexus[UInt](
+    inputFn = BundleBridgeNexus.orReduction[UInt](p(HartPrefixKey)) _,
+    outputFn = (prefix: UInt, n: Int) =>  Seq.tabulate(n) { i =>
+      val y = dontTouch(prefix | hartIdList(i).U(p(MaxHartIdBits).W))
+      if (p(HartPrefixKey)) BundleBridgeNexus.safeRegNext(y) else y
+    },
+    default = Some(() => 0.U(p(MaxHartIdBits).W))
+  )
+  // TODO: this also needs to be wired into the DebugModule's hart selection circuit?
+
+  // Collect a reset vector address and distribute to the tiles
+  val tileResetVectorNode = BundleBridgeEphemeralNode[UInt]()
+  val tileResetVectorNexusNode = BundleBroadcast[UInt]()
+
+  // Create subsystem IOs that are based on the number of tiles
+  //   or, if such IOs don't exist, rely on connections of drivers to internal nexus nodes
+  val tileHartIdIONodes: Seq[BundleBridgeSource[UInt]] = p(SubsystemExternalHartIdWidthKey) match {
+    case Some(w) => Seq.fill(tiles.size) {
+      val hartIdSource = BundleBridgeSource(() => UInt(w.W))
+      tileHartIdNode := hartIdSource
+      hartIdSource
+    }
+    case None => { tileHartIdNode :*= tileHartIdNexusNode; Nil }
+  }
+
+  val tileResetVectorIONodes: Seq[BundleBridgeSource[UInt]] = p(SubsystemExternalResetVectorKey) match {
+    case true => Seq.fill(tiles.size) {
+      val resetVectorSource = BundleBridgeSource[UInt]()
+      tileResetVectorNode := resetVectorSource
+      resetVectorSource
```

Comment:
same Q, should there be an InModuleBody check to make sure no one is trying to connect to tileResetVectorNexusNode?